### PR TITLE
docs: cluster-configuration: fix the network annotation

### PR DESCRIPTION
**Pull request by dankenigsberg**
This PR fixes our doc by using the correct annotation used to affect underlying CNAO operator. It also prettifies the formatting of annotation examples.

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [x] PR Message
- [x] Commit Messages
- [x] How to test
- [x] Unit Tests
- [x] Functional Tests
- [x] User Documentation
- [x] Developer Documentation
- [x] Upgrade Scenario
- [x] Uninstallation Scenario
- [x] Backward Compatibility
- [x] Troubleshooting Friendly
  
**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by nunnatsa — COMMENTED** [review]
Thank you for improving our documentation. Added two small comments.

File: docs/cluster-configuration.md [hunk 1]
```diff
@@ -589,13 +589,15 @@ HCO enables users to modify the operand CRs directly using jsonpatch annotations
 Modifications done to CRs using jsonpatch annotations won't be reconciled back by HCO to the opinionated defaults.  
 The following annotations are supported in the HyperConverged CR:
 * `kubevirt.kubevirt.io/jsonpatch` - for KubeVirt configurations
-* `containerizeddataimporter.kubevirt.io/jsonpatch` - for CDI configurations
-* `networkaddonsconfig.kubevirt.io/jsonpatch` - for CNAO configurations
+* `containerizeddataimporter.kubevirt.io/jsonpatch` - for [CDI](https://github.com/kubevirt/containerized-data-importer/) configurations
```
[the remark this answers is not in the document]
Comment (dankenigsberg) [reply docs/cluster-configuration.md:592]: Makes sense. The best proof that a URL is needed is that I don't know where is the spec of the Kind CDI. Can you help me find it @nunnatsa ?

File: docs/cluster-configuration.md [hunk 2]
```diff
@@ -589,13 +589,15 @@ HCO enables users to modify the operand CRs directly using jsonpatch annotations
 Modifications done to CRs using jsonpatch annotations won't be reconciled back by HCO to the opinionated defaults.  
 The following annotations are supported in the HyperConverged CR:
 * `kubevirt.kubevirt.io/jsonpatch` - for KubeVirt configurations
```
[the remark this answers is not in the document]
Comment (dankenigsberg) [reply docs/cluster-configuration.md:591]: hmmm, here too I find [explanations for specific fields](https://kubevirt.io/user-guide/operations/live_migration/#changing-cluster-wide-migration-limits) but not a complete reference of Kind KubeVirt.

File: docs/cluster-configuration.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (nunnatsa) [reply docs/cluster-configuration.md:592]: KubeVirt API: https://github.com/kubevirt/api. There is also the API reference: https://kubevirt.io/api-reference/
CDI API: https://github.com/kubevirt/containerized-data-importer-api and the API reference: https://kubevirt.io/cdi-api-reference/

[the remark this answers is not in the document]
Comment (dankenigsberg) [reply docs/cluster-configuration.md]: Ouch, KubeVirt and CDI's are quite hard to read, but that goes beyond this PR.

**Review by nunnatsa — APPROVED** [review]
/approve

**Pull request merged**